examples/chip-tool/templates/partials/StructLoggerImpl.zapt

```diff
@@ -3,16 +3,37 @@ CHIP_ERROR DataModelLogger::LogValue(const char * label, size_t indent, const ch
   DataModelLogger::LogString(label, indent, "{");
 {{#zcl_struct_items}}
   {
+{{#if (isEqual label "DeviceType") }}
+{{#if (isEqual parent.name "DeviceTypeStruct") }}
```

Comment:
Would it work to check that the item's type is "devtype-id" instead of doing this name hardcoding?

Similarly, nullability could be checked on the item.

Because this template as written will generate non-compiling code as soon as there is an item named DeviceType that is not nullable in a struct that is not named DeviceTypeStruct.

Reply:
Done.